### ISSUE TITLE
Style/FormatString should detect hash literal on left hand side of `%`

### ISSUE THREAD

**Issue by backus** [bbatsov/rubocop#3839]
See this thread: https://github.com/bbatsov/rubocop/pull/3837#discussion_r94213113

The `FormatString` cop should detect this syntax and recommend `Kernel#format` over `String#%`:

```ruby
message = MSG % {
  name: identifier,
  origin: origin,
  options: options.join(' or ')
}
```

**Comment by backus** [comment]
@Drenmi no that is about using `%<foo>s` vs `%s` vs `%{foo}`. This is about `msg % { foo: 'abc' }` vs `msg % ['abc']` vs `format(msg, foo: 'abc')` vs `format(msg, ['abc'])`
